File: selectionsort.cpp
#include <iostream>
using namespace std;

int main() {
  srand(time(NULL));

  int nums[10];
  for (int i = 0; i < 10; i++){
    nums[i] = rand() % 10;
  }

  cout << "Lista original: ";
  for (int i = 0; i < 10; i++){
    cout << nums[i] << " ";
  }
  cout << endl;

  
  for (int i = 0; i < 10; i++){
    bool check = false;
    int index;
    int tmp = nums[i];
    for (int j = i + 1; j < 10; j++){
      if (nums[j] < tmp){
        tmp = nums[j];
        check = true;
        index = j;
      }
    }
    if (check){
      nums[index] = nums[i];
      nums[i] = tmp;
    }
  }
  
  cout << "Lista ordenada: ";
  for (int i = 0; i < 10; i++){
    cout << nums[i] << " ";
  }
}

File: selectionsort.py
# Pseudocode
# Inicia en un pivote, el primero de la lista
# Itera aumentando en una posición el pivote
#   Para cada iteración, encuentra el menor/mayor número
#    Intercambia el menor/mayor con el pivote
#    Regresa

# Declaramos la lista
lista = [0,1,2,3,4,5,6,7,8,9]

# Declaramos la función selectionsort
def selectionsort(lista):
  for i in range(0, len(lista)):
    bandera = False
    piv = lista[i]
    index = 0
    for j in range(i + 1, len(lista)):
      if (piv < lista[j]):
        piv = lista[j]
        bandera = True
        index = j
    
    if (bandera):
      tmp = lista[i]
      lista[i] = lista[index]
      lista[index] = tmp
        

# Función para imprimir valores en una lista
def imprimirLista(lista):
  for i in lista:
    print(i, " ", end="")
    
print("Lista desordenada: ", end="")
imprimirLista(lista)
print()

# Llamamos a la función selectionsort
selectionsort(lista)

print("Lista ordenada: ", end="") 
imprimirLista(lista)


